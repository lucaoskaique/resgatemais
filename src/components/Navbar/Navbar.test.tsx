import { render, screen } from "@testing-library/react"
import { describe, expect, it } from "vitest"

import { Navbar } from "."

describe("<Navbar />", () => {
  it('should render the "Sobre" link', () => {
    render(<Navbar />)
    expect(screen.getByTestId("sobre-link-small")).toBeDefined()
    expect(screen.getByTestId("sobre-link-large")).toBeDefined()
  })

  it('should render the "Ações" link', () => {
    render(<Navbar />)
    expect(screen.getByTestId("acoes-link-small")).toBeDefined()
    expect(screen.getByTestId("acoes-link-large")).toBeDefined()
  })
})
