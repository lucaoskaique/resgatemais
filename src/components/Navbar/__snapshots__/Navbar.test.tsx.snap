// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Navbar /> > should match the snapshot 1`] = `
<div>
  <nav>
    <ul
      class="flex itens-center gap-9"
    >
      <li
        class="py-3"
      >
        <a
          href="#"
        >
          Sobre
        </a>
      </li>
      <li
        class="py-3"
      >
        <a
          href="#"
        >
          Ações
        </a>
      </li>
      <li>
        <button>
          <span
            class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                aria-label="Switch demo"
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
        </button>
      </li>
      <li
        class="py-3"
      >
        <a
          class="bg-color-red px-4 py-3 rounded-[0.313rem]"
          href="#"
        >
          Doar
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<Navbar /> > should render the heading 1`] = `
<body>
  <div>
    <div>
      <h1>
        Navbar
      </h1>
    </div>
  </div>
</body>
`;
